x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/core/security/transport/netty4/SecurityNetty4Transport.java

```diff
@@ -143,6 +149,78 @@ protected ChannelHandler getClientChannelInitializer(DiscoveryNode node, Connect
         return new SecurityClientChannelInitializer(node, connectionProfile);
     }
 
+    @Override
+    protected Bootstrap getClientBootstrap(ConnectionProfile connectionProfile) {
+        final Bootstrap bootstrap = super.getClientBootstrap(connectionProfile);
+        if (false == REMOTE_CLUSTER_PROFILE.equals(connectionProfile.getTransportProfile())
+            || remoteClusterClientBootStrapOptions.isEmpty()) {
+            return bootstrap;
+        }
+
```

Comment:
I considered it and decided to keep it mostly as is. I did make a few improvements:
1. Moved the re-configuration code from `getClientBootstrap` into `RemoteClusterClientBootstrapOptions.configure` (akin to your suggestion of `apply`). I think this change makes the Options class more useful and looks better. 
2. Added assertions in `SimpleSecurityNetty4ServerTransportTests.testGetClientBootstrap` to ensure the options of the default clientBootstrap are the known and expected. This is to detect possible future change to how options are configured in `Netty4Transport.createClientBootstrap`.
3. Added a comment in `Netty4Transport.createClientBootstrap` to say any change to this method should also check whether it is also needed in `RemoteClusterClientBootstrapOptions.configure`.

There are a few reasons why I chose to keep the existing changes. The most important one is that `Netty4Transport.createClientBootstrap` is for configuring a new Bootstrap while `RemoteClusterClientBootstrapOptions.configure` is to re-configure an existing Bootstrap. The later has logics, e.g. remove the option if it is `null`, that are unnecessary and heavy weighted for the former use case. It would make the otherwise simple `createClientBootstrap` method harder to understand. Alternatively, We could also make `Nett4Transport` class to also go through the "create" and "reconfigure" process like the security version. But that means the logics are still duplicated between "create" and "reconfigure". So it does not feel much better. Overall, I prefer to keep `Netty4Transport` simple since it is more common code path.